packages/core-flows/src/definition/cart/steps/get-actions-to-compute-from-promotions.ts

```diff
@@ -0,0 +1,54 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { CartDTO, IPromotionModuleService } from "@medusajs/types"
+import { deduplicate, isString } from "@medusajs/utils"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+interface StepInput {
+  cart: CartDTO
+  promoCodes: string[]
+  removePromotions: boolean
+}
+
+export const getActionsToComputeFromPromotionsStepId =
+  "get-actions-to-compute-from-promotions"
+export const getActionsToComputeFromPromotionsStep = createStep(
+  getActionsToComputeFromPromotionsStepId,
+  async (data: StepInput, { container }) => {
```

Comment:
thought(non-blocking): Would it make sense for this step to be even more generic i.e. just accept `codes` and return the computed actions? All the data extraction can be performed in transformers. Wdyt?

Reply:
Can you elaborate on this one? I don't fully understand.

For us to be able to generate actions to compute, we need the full cart along with the promo codes.

we also need a third parameter - `action`, which can be:

- `add`: adds new promo codes along with already applied promo codes in the cart (case: add promo code to cart)
- `remove`: removes only one promo code from the cart, keeps all existing promo codes (case: remove promo code from cart)
- `replace`: replace the given promo codes, i.e. removes all existing ones and add only the ones in this given promo code array (case: update cart)

The promo codes to pass into the compute actions will vary depending on each of these actions.

Reply:
Yes, I am not sure what I was thinking actually. I take it back :D